src/workflow-engine/charms/temporal-worker/oci_factory/activities/activity_consume_events.py

```diff
@@ -61,6 +74,11 @@ async def consume(topic: str, consumer_group: str) -> dict:
     proc = subprocess.Popen(
         ["python3", script_full_path, "{}".format(value.get("release"))]
     )
-    proc.wait()
+    ret_code = proc.wait()
+
+    status = "Success" if ret_code == 0 else "Failed"
```

Comment:
For this, can it be a good idea to use the walrus operator to avoid  doing two times the if condition?

source: https://realpython.com/python-walrus-operator/

Reply:
Thanks for the suggestion. I found directly using `success = proc.wait() == 0` is more concise. Fixed in [2674da7](https://github.com/canonical/oci-factory/commit/2674da7918c9f2eeeb71b88e1e7bf506e31498ba).